modules/renter/persist.go

```diff
@@ -174,26 +175,26 @@ func (f *file) UnmarshalSia(r io.Reader) error {
 }
 
 // saveFile saves a file to the renter directory.
-func (r *Renter) saveFile(f *file) error {
-	if f.deleted {
+func (r *Renter) saveFile(f *siafile.SiaFile) error {
+	if f.Deleted() { // TODO: violation of locking convention
```

Comment:
hmm. I'm not sure there's an easy path to fix this.

Typically you'd just turn `saveFile` into `managedSaveFile`, but I think that there are places where we need to keep holding the lock before and after we call `saveFile` to be safe.